acceptance/tests/test_upgrade_puppet5_to_puppet6.rb

```diff
@@ -2,28 +2,21 @@
 require_relative '../helpers'
 
 # Tests FOSS upgrades from the latest puppet 5 (the puppet5 collection) to puppet 6.0.0.
-# This test will only run if the version of puppet on the master host is less than 6.
 test_name 'puppet_agent class: collection parameter for FOSS upgrades' do
-  master_agent_version = puppet_agent_version_on(master)
+  master_collection restrict_to: 'puppet5'
```

Comment:
There are a few things in these updated tests that look like readability problems to me:

- It's not obvious to me from the name that `is_foss_ugprade_test` is restricting the upgrade from PE platforms -- that name sounds like a boolean check of some kind. I'd prefer something like `exclude_pe_upgrade_platforms` or similar that describes the action it's doing.
- Likewise, I'd prefer a verb-oriented method name to replace `master_collection`. I'd prefer something like `require_master_collection('puppet5')`, `require_master_collection(max: 'puppet5')`, or similar.
- It seems like `assert_successful_upgrade` would be better titled `assert_agent_version` or similar. A 'successful upgrade' may entail more than just updating the puppet-agent version: we may want to assert that certain services are stopped or running depending on the parameters that were used, too.
- This is more of a nitpick, but the words 'setup', 'set-up', and 'set up' are all being used here. 'set up' is the verb, and 'setup' is the noun version -- I'd prefer these to be consistent.

I am not a fan of the lambdas, which are not very ruby-like and unfamiliar to people who are used to writing normal beaker tests already, but it's not a big deal.

Reply:
Updated the PR to address your concerns. Here are the changes:

* Changed `is_foss_upgrade_test` => `exclude_pe_upgrades`
* Changed to `require_master_collection` as you specified -- that is cleaner.
* `assert_successful_upgrade` now takes a block so you can do something like
```
assert_succesful_upgrade('6.0.0') do |agent|
  // additional assertions
end
```
* Addressed the set-up (verb) vs. setup (noun) inconsistency
* Updated the commit to reference MODULES-8348